src/functions-and-arrays.js

```diff
@@ -1,41 +1,122 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(num1, num2) {
+  if (num1 > num2) {
+    return num1;
+  }
+  else {
+    return num2;
+  }
+}
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
-
+function findLongestWord(words) {
+  let longestWord = ' ';
+  for ( let i = 0; i < words.length; i++){
+    if(words[i].length > longestWord){
+	longestWord = words[i];
+     }
+  return longestWord;
+  }
+} 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
+function sumNumbers(numbers) {
+  let sum = 0;
+  for (let i = 0; i < numbers.length; i++) {
+      sum += numbers[i];
+  }
+  return sum;
+}
+
 
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
+function sum(mixedArr) {
+  let total = 0;
+
+    for (let i = 0; i < mixedArr.length; i++) {
+        switch (typeof mixedArr[i]) {
+            case 'number':
+                total += mixedArr[i];
+                break;
+
+            case 'string':
+                total += mixedArr[i].length;
+                break;
+
+            case 'boolean':
+                if (mixedArr[i] === true) {
+                    total++;
+                }
+                break;
+        }
+    }
+
+    return total;
+}
+
+  
 
```

Comment:
Super gut, dass du auch Bonus Aufgaben gemacht hast. Wenn du noch genauer in die Lösung interessiert bist, sie ist wieder für euch im Student Portal verfügbar!